src/app/tree-grid/tree-grid-remote-filtering-sample/tree-grid-remote-filtering-sample.component.ts

```diff
@@ -0,0 +1,39 @@
+import { AfterViewInit, ChangeDetectorRef, Component, OnInit, ViewChild } from "@angular/core";
+import { IgxToastComponent, IgxTreeGridComponent } from "igniteui-angular";
+import { Observable } from "rxjs";
+import { RemoteFilteringService } from "../services/remoteFilteringService";
+
+@Component({
+    providers: [RemoteFilteringService],
+    selector: "app-tree-grid-remote-filtering-sample",
+    styleUrls: ["./tree-grid-remote-filtering-sample.component.scss"],
+    templateUrl: "./tree-grid-remote-filtering-sample.component.html"
+})
+export class TreeGridRemoteFilteringSampleComponent implements OnInit, AfterViewInit {
+
+    public remoteData: Observable<any[]>;
+    @ViewChild("treeGrid") public treeGrid: IgxTreeGridComponent;
+    @ViewChild("toast") public toast: IgxToastComponent;
+
+    constructor(private _remoteService: RemoteFilteringService, private _cdr: ChangeDetectorRef) {
+    }
+
+    public ngOnInit() {
+        this.remoteData = this._remoteService.remoteData;
+    }
+
+    public ngAfterViewInit() {
+        this.processData();
+        this._cdr.detectChanges();
```

Comment:
This change detection seems unnecessary. You could also remove the _cdr from the constructor.

Reply:
In stackblitz, the sample throws an error that requires invoking detectChanges in afterViewInit.